xscen/catutils.py

```diff
@@ -252,7 +252,7 @@ def _name_parser(
     }
 
 
-def _parse_dir(
+def _parse_dir(  # noqa: C901
```

Comment:
We should probably go for your 2nd option, then. 

Many of our functions will be inherently "complex", because we need to support many possible use cases and, most of the time, things aren't really reusable and thus aren't worth moving to helper functions.